torch/_functorch/_aot_autograd/schemas.py

```diff
@@ -117,6 +117,11 @@ class InputAliasInfo:
     mutates_storage_metadata: bool
     requires_grad: bool
     keep_input_mutations: bool
+    # JointFn Mutation Info, is filled only after jointFn tracing.
+    joint_mutates_data: Optional[bool] = None
```

Comment:
We only actually need to support `joint_mutates_data` - if any of the other 3 types of mutation below happen during the backward, we should raise an error during tracing and say we don't support it (we technically could support it, but metadata mutation / storage mutation are all kind of niche, and supporting them in the backward adds complexity that we probably don't care about right now)

Reply:
If these are only going to be set when we need autograd (i.e. trace_joint = True), what do you think about keeping them in an optional list of indices or bools instead of directly on InputInfo? That way you can avoid needing to assert that the joint_mutates_* is not none each time.

Reply:
Agreed, let's just make it a (non-optional?) list of indices, similar to `indices_of_inps_to_detach` ([code](https://github.com/pytorch/pytorch/blob/main/torch/_functorch/_aot_autograd/runtime_wrappers.py#L130)).

This is actually pretty important for runtime perf: models can have 100+ inputs to the forward and backwrd graph, and looping through all 100 of them can have noticeable runtime overhead.

we can add index `i` to the list only if input i gets a backward mutation **and** it requires grad. That way, our assertion at runtime only has to check if the list is non-empty to raise an error ( but it can still print the actual indices for a good error message).